vertical-pod-autoscaler/pkg/recommender/input/controller_fetcher/controller_fetcher_test.go

```diff
@@ -239,17 +272,42 @@ func TestControllerFetcher(t *testing.T) {
 			expectedKey:   nil,
 			expectedError: fmt.Errorf("Cycle detected in ownership chain"),
 		},
+		{
+			key: &ControllerKeyWithAPIVersion{ControllerKey: ControllerKey{
+				Name: "test-deployment", Kind: "Deployment", Namespace: "test-namesapce"}},
+			objects: []runtime.Object{&appsv1.Deployment{
+				TypeMeta: metav1.TypeMeta{
+					Kind: "Deployment",
+				},
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "test-deployment",
+					Namespace: "test-namesapce",
+					// UnkownParent
+					OwnerReferences: []metav1.OwnerReference{
+						{
+							APIVersion: "Foo/Foo",
+							Controller: &trueVar,
+							Kind:       "Foo",
+							Name:       "bah",
+						},
+					},
+				},
+			}},
+			expectedKey: &ControllerKeyWithAPIVersion{ControllerKey: ControllerKey{
+				Name: "test-deployment", Kind: "Deployment", Namespace: "test-namesapce"}}, // No known parent or a parent that can scale so should return itself
```

Comment:
I understand this case as "Parent does not support scale subresource so should return itself"